lm_eval/tasks/redlite/utils.py

```diff
@@ -0,0 +1,103 @@
+import re
+import string
+from rouge_score import rouge_scorer, scoring
+
+
+# Input example 
+# {
+#   "messages": [
+#     {
+#       "role": "system",
+#       "content": "You are a helpful problem solving assistant able to determine if a problem statement has enough information to find a solution. When you have enough information to solve a problem, you answer with \"Yes, this problem can be solved with provided information.\". If information required to solve is incomplete, unavailable or missing you answer with \"No, information is missing in order to solve this problem.\"."
+#     },
+#     {
+#       "role": "user",
+#       "content": "Here is a problem statement. Determine if it can be solved. Answer stricly with \"Yes, this problem can be solved with provided information.\" or \"No, information is missing in order to solve this problem.\" depending if you have enough information to solve or not.\nFor instance, if the problem was \"Natalia sold clips to 48 of her friends in April, and then she sold half as many clips in May. How many clips did Natalia sell altogether in April and May?\", your answer would be: \"Yes, this problem can be solved with provided information.\".\nConversely, if the problem was \"A paper about AI regulation that was originally submitted to arXiv.org in June 2022 shows a figure with three axes, where each axis has a label word at both ends. Which of these words is used to describe a type of society in a Physics and Society article submitted to arXiv.org on August 11, 2016?\", your answer would be: \"No, information is missing in order to solve this problem.\".\nHere's the problem statement: \"Sandra's neighbor gives her a basket of 9 eggs every time she babysits their daughter. To make a Spanish flan, she needs 3 eggs. If Sandra has been tasked to make 15 Spanish flans for her school fundraiser, how many times does Sandra have to babysit?\".\nCan it be solved?\n"
+#     }
+#   ],
+#   "expected": "Yes, this problem can be solved with provided information.",
+#   "id": "test-0"
+# }
+
+def doc_to_text(doc):
+    return doc["messages"][1]["content"]
+
+def process_results(doc, results):
+
+    completion = results[0]
+    
+    pem_score = calculate_pem(doc["expected"], completion)
+    rouge_score = calculate_rouge(doc["expected"], completion)
+    
+    return {
+        "best-of-pem-rouge" : max(pem_score, rouge_score)
+    }
+    
+
+
+def calculate_rouge(gold: str, pred: str) -> float:
+    scorer = rouge_scorer.RougeScorer(["rouge2"], use_stemmer=True)

```

Comment:
Could we carry over the change made in #2090 to prevent excessive logs from RougeScorer?

Reply:
done